src/Analyzers/CSharp/Analyzers/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,113 @@
+                return;
+
+            Recurse(context, option.Notification.Severity, context.Tree.GetRoot(context.CancellationToken));
+        }
+
+        private void Recurse(SyntaxTreeAnalysisContext context, ReportDiagnostic severity, SyntaxNode node)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (node is ConditionalExpressionSyntax conditionalExpression)
+                ProcessConditionalExpression(context, severity, conditionalExpression);
+
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.IsNode)
+                    Recurse(context, severity, child.AsNode()!);
+            }
+        }
+
+        private void ProcessConditionalExpression(
+            SyntaxTreeAnalysisContext context, ReportDiagnostic severity, ConditionalExpressionSyntax conditionalExpression)
+        {
+            // Don't bother analyzing nodes that have syntax errors in them.
+            if (conditionalExpression.GetDiagnostics().Any(static d => d.Severity == DiagnosticSeverity.Error))
+                return;
+
+            // Only if both tokens are not ok do we report an error.  For example, the following is legal:
+            //
+            //  var x =
+            //      goo ? bar :
+            //      baz ? quux : ztesh;
+            //
+            // despite one colon being at the end of the line.
+            if (IsOk(conditionalExpression.QuestionToken) ||
+                IsOk(conditionalExpression.ColonToken))
+            {
+                return;
+            }
+
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                this.Descriptor,
+                conditionalExpression.QuestionToken.GetLocation(),
+                severity,
+                additionalLocations: null,
+                properties: null));
+
+            return;
+
+            static bool IsOk(SyntaxToken token)
+            {
+                // Only care about tokens that are actually present.  Missing ones mean the code is incomplete and we
+                // don't want to complain about those.
+                if (token.IsMissing)
+                    return true;
+
+                // question/colon has to be at the end of the line for us to actually care.
+                if (token.TrailingTrivia is not [.., SyntaxTrivia(SyntaxKind.EndOfLineTrivia)])
+                    return true;
+
+                // if the next token has pp-directives on it, we don't want to move the token around as we may screw
+                // things up in different pp-contexts.
+                var nextToken = token.GetNextToken();
+                if (nextToken == default)
+                    return true;
+
+                if (nextToken.LeadingTrivia.Any(static t => t.Kind() is
+                        SyntaxKind.IfDirectiveTrivia or SyntaxKind.ElseDirectiveTrivia or SyntaxKind.ElifDirectiveTrivia or SyntaxKind.EndIfDirectiveTrivia))
```

Comment:
Should this be extracted into a helper method in syntax facts?

Reply:
will likely do in followup as i use this as well for the next formatting option.